### PR TITLE
Core/Scripting: add QuestScript with 2 first hooks

### PR DESCRIPTION
**Pull request by Traesh**
**Changes proposed:**

-  DB linked script for quests

**Target branch(es):**

- master

**Tests performed:** (Does it build, tested in-game, etc.)
No test performed yet

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] Add more hooks ?

### CODE REVIEW

**Comment by funjoker** [comment]
OnQuestComplete
OnQuestReward
OnQuestObjectiveComplete
OnQuestObjectiveProgress(Some objectives have a bar with % filled up specifying the amount which should be reached to get triggered as parameter)
OnQuestFailed
OnQuestAccept(maybe a non NPC/GO/ITEM/SPELL bound, general way)
OnQuestAbandon

These are prolly all hooks which could be useful ^^

**Comment by Seyden** [comment]
We only need 1 hook. OnQuestStatusChange... then you can check which is the new status..

**Comment by funjoker** [comment]
You never rly fix quest zones. I say it's useful to have those

**Comment by Seyden** [comment]
it doesnt make any sense to implement this hooks.. you have OnQuestStatusChanged.. what do you want more? This hook is a bundle of all these hooks that you want

**Comment by Shauren** [comment]
I agree with @Seyden - these 2 new hooks give you old and new statuses as arguments (and are called everywhere)

**Review by Shauren — APPROVED** [review]

File: src/server/game/Scripting/ScriptMgr.cpp [hunk 1]
```diff
@@ -748,6 +752,35 @@ class ScriptRegistrySwapHooks<SceneScript, Base>
     bool swapped;
 };
 
+/// This hook is responsible for swapping QuestScript's
```
Comment (joschiwald) [inline src/server/game/Scripting/ScriptMgr.cpp]: can you explain when we need these ScriptRegistrySwapHooks? i was thinking that we don't need such hooks for not "living" objects and scripts not bound to any object

Comment (malcrom) [reply src/server/game/Scripting/ScriptMgr.cpp]: Traesh, can you respond to this?

Comment (Traesh) [reply src/server/game/Scripting/ScriptMgr.cpp]: I think joschiwald is right, maybe @Naios could confirm

Comment (Naios) [reply src/server/game/Scripting/ScriptMgr.cpp]: Yes, the hook isn't needed here.
It's only needed for scripts with a custom unload logic such as database bound scripts.

Comment (Traesh) [reply src/server/game/Scripting/ScriptMgr.cpp]: Well this script is database bound, it's just not entity linked

Comment (Naios) [reply src/server/game/Scripting/ScriptMgr.cpp]: @Traesh Yes, you are right, my bad. The code is fine.
I was just wondering the fact that you only implemented the assertions,
but this is equal to the hooks of `InstanceMapScript` and maybe those could be merged in the future.
:+1:

**Pull request merged**